# Конспект от 3 февраля 2023 по теме:
### Возбуждение исключений
Для возбуждения исключения используется оператор **raise**.

В качестве аргумента оператор **raise** использует экземпляр класса, унаследованного от Exception.

Класс указывает на тип исключения, а аргумент, передаваемый конструктору этого исключения, обычно описывает подробности возникновения исключительной ситуации.

Если возбуждаемые пользователем исключения не перехватываются, то они распространяются вплоть до стандартного обработчика исключений и прекращают работу программы с выводом стандартного сообщения об ошибке.

> Возбуждаемые пользователем исключения перехватываются тем же способом, что и исключения, которые возбуждает интерпретатор Python.

###### Приведённый ниже код:
```
try:
	raise IndexError('Ошибка в коде')
except Exception as err:
	print(err)
	print(type(err))
```
###### Выведет:
```
Ошибка в коде
<class 'IndexError'>
```
###### Пример 2:
```
def get_month_name(index):
	try:
		if not str(index).isdigit():
			raise TypeError('Аргумент должен быть числом.')
		if index < 1 or index > 12:
			raise ValueError('Аргумент должен быть числом от 1 до 12.')
	except Exception as err:
		return err
	else:
		return index
print(get_month_name(25))
print(get_month_name('строка'))
print(get_month_name(8))
```
###### Выведет:
```
Аргумент должен быть числом от 1 до 12.
Аргумент должен быть числом.
8
```

При создании объекта исключения можно указать в конструкторе подробности возникновения исключительной ситуации.

Конструктор любого исключения принимает переменное количество аргументов.

Для того чтобы получить указанные значения, используется атрибут **args**, который является кортежем.

###### Приведённый ниже код:
```
try:
	x, y = 10, 0
	if y == 0:
		raise ZeroDivisionError('Произошло деление на ноль.')
except ZeroDivisionError as err:
	print(err)
	print(err.args)
	print(type(err.args))
```
###### Выведет:
```
Произошло деление на ноль.
('Произошло деление на ноль.',)
<class 'tuple'>
```
> При печати объекта исключения автоматически происходит вызов функции str().

Оператор возбуждения исключений raise имеет несколько вариантов формата:

###### Вариант 1:
